tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -178,32 +177,43 @@ stages:
         - task: DownloadPipelineArtifact@2
           displayName: Download test files
           inputs:
+            # It seems there's a bug and preferTriggeringPipeline is not respected.
+            # We force the behavior by explicitly specifying:
+            # - buildVersionToDownload: specific
+            # - buildId: <the id of the triggering build>
+            # preferTriggeringPipeline: true
             source: specific
             project: internal
-            pipeline: ${{ variables.artifactPipeline }}
-            preferTriggeringPipeline: true
-            allowPartiallySucceededBuilds: true
-            runVersion: latestFromBranch
-            runBranch: $(Build.SourceBranch)
+            pipeline: $(resources.pipeline.client.pipelineName)
+            buildVersionToDownload: specific
+            buildId: $(resources.pipeline.client.runID)
```

Comment:
`buildVersionToDownload` corresponds to `runVersion` in the old code. `branchName` (commented, below) corresponds to `runBranch`. I thought these aliases were more descriptive and understandable when figuring out what the task is doing.